tracing-appender/src/lib.rs

```diff
@@ -1,15 +1,80 @@
+//! Writers for logging events/spans
+//!
+//! # Overview
+//!
+//! [`tracing`][tracing] is a framework for instrumenting Rust programs to collect
+//! structured, event-based diagnostic information. This crate provides the ability
+//! for `tracing` events and spans to be recorded in a non blocking manner by using a
+//! dedicated logging thread. It also provides a [`RollingFileAppender`][file_appender]
+//! that can be used with or without the non blocking writer.
+//!
+//! [file_appender]: ./rolling/struct.RollingFileAppender.html
+//! [tracing]: https://docs.rs/tracing/
+//!
+//! # Usage
+//! This crate can be used in a few ways to record spans/events:
+//! 1. Using a `RollingFileAppender` to perform writes to a log file. This will block on writes.
+//! 2. Using *any* `std::io::Write` implementation in a non-blocking way
+//! 2. Using a combination of `NonBlocking` and `RollingFileAppender` to allow writes to a log file
+//! without blocking.
+//!
+//! ## Rolling File Appender
+//!
+//! ```rust
+//! # fn docs() {
+//! use tracing_appender::rolling::{RollingFileAppender, Rotation};
+//!
+//! let file_appender = RollingFileAppender::new(Rotation::HOURLY, "/some/directory", "prefix.log");
+//! # }
+//! ```
+//! This creates an hourly rotating file appender which outputs to `/some/directory/prefix.log.YYYY-MM-DD-HH`.
+//! <br/> [`Rotation::DAILY`] and [`Rotation::NEVER`] are the other available options.
+//!
+//! It implements the `std::io::Write` trait. To use with a subscriber, it must be combined with a
+//! [`MakeWriter`][make_writer] implementation to be able to record tracing spans/event.
+//!
+//! [make_writer]: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/trait.MakeWriter.html
+//!
+//! The [`rolling module`][rolling]'s documentation provides more detail on how to use this file appender.
+//!
+//! [rolling]: ./rolling/index.html
+//!
+//! ## Non Blocking Writer
+//!
+//! ```rust
+//! # fn doc() {
+//! let (non_blocking, _guard) = tracing_appender::non_blocking(std::io::stdout());
+//! let subscriber = tracing_subscriber::fmt().with_writer(non_blocking);
+//! tracing::subscriber::set_global_default(subscriber.finish()).expect("Could not set global default");
+//! # }
+//! ```
+//!
+//! Alternatively, you can provide *any* type that implements `std::io::Write` to `tracing_appender::nonblocking`
+//!
+//! The [`non_blocking module`][non_blocking]'s documentation provides more detail on how to use `non_blocking`.
+//!
+//! [non_blocking]: ./non_blocking/index.html
+//!
+//! ## Non Blocking and rolling file appender
+//!
+//! ```rust
+//! # fn docs() {
+//! use tracing_appender::rolling::{RollingFileAppender, Rotation};
+//!
+//! let file_appender = RollingFileAppender::new(Rotation::HOURLY, "/some/directory", "prefix.log");
```

Comment:
may be nicer to show
```suggestion
//! let file_appender = tracing_appender::rolling::hourly("/some/directory", "prefix.log");
```